examples/src/main/scala/caliban/play/CalibanController.scala

```diff
@@ -0,0 +1,42 @@
+package caliban.play
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import caliban.{play => _, _}
+import caliban.ExampleService.ExampleService
+import play.api.mvc._
+import zio.{ Runtime, ZEnv, ZLayer }
+
+class CalibanController(val controllerComponents: ControllerComponents)(
+  implicit actorSystem: ActorSystem,
+  materializer: Materializer
+) extends BaseController {
+
+  val calibanPlayAdapter: PlayAdapter =
+    PlayAdapter(controllerComponents.parsers, controllerComponents.actionBuilder)
+
+  implicit val runtime = Runtime.default
+  implicit val ec      = actorSystem.dispatcher
+
+  val service: ZLayer[Any, Nothing, ExampleService] = ExampleService.make(ExampleData.sampleCharacters)
+
+  val interpreter: GraphQLInterpreter[ZEnv, CalibanError] = runtime.unsafeRun(
+    ExampleService
+      .make(ExampleData.sampleCharacters)
+      .memoize
+      .use(layer => ExampleApi.api.interpreter.map(_.provideCustomLayer(layer)))
+  )
```

Comment:
Usually I move the `layer` into the runtime, then bind the shutdown hook to the runtime shutdown, which cleans up the resources, but that is a matter of preference I suppose.